packages/dds/tree/src/test/shared-tree/treeApi.spec.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "node:assert";
+import { Tree } from "../../shared-tree/index.js";
+import { SchemaFactory, TreeConfiguration } from "../../simple-tree/index.js";
+import { getView } from "../utils.js";
+
+const schema = new SchemaFactory("com.example");
+class TestObject extends schema.object("TestObject", { content: schema.number }) {}
+
+describe("treeApi", () => {
+	describe("runTransaction invoked via a tree view", () => {
+		it("runs transactions", () => {
+			const view = getView(new TreeConfiguration(TestObject, () => ({ content: 42 })));
+			Tree.runTransaction(view, (root) => {
+				root.content = 43;
+			});
+			assert.equal(view.root.content, 43);
+		});
+
+		it("can be rolled back", () => {
+			const view = getView(new TreeConfiguration(TestObject, () => ({ content: 42 })));
+			Tree.runTransaction(view, (root) => {
+				root.content = 43;
+				return "rollback";
+			});
+			assert.equal(view.root.content, 42);
+		});
+
+		it("rolls back transactions on error", () => {
+			const view = getView(new TreeConfiguration(TestObject, () => ({ content: 42 })));
+			try {
+				Tree.runTransaction(view, (root) => {
+					root.content = 43;
+					throw Error("Oh no");
+				});
+			} catch (e) {
+				assert(e instanceof Error);
+				assert.equal(e.message, "Oh no");
+			}
+			assert.equal(view.root.content, 42);
+		});
+
+		// TODO: Either enable when afterBatch is implemented, or delete if no longer relevant
+		it.skip("emits change events", () => {
```

Comment:
why does this test use afterBatch?

Reply:
In this case, the user has the `TreeView` rather than just the node, and is using the view directly. They could listen for the events with `Tree.on(view.root)` but since test in the other describe already does that, it seemed a good opportunity to check that events are getting emitted by the view at the top level as well. Although, they aren't :0